app/models/podcast_import.rb

```diff
@@ -0,0 +1,267 @@
+# encoding: utf-8
+
+require 'prx_access'
+require 'addressable/uri'
+require 'feedjira'
+require 'itunes_category_validator'
+
+class PodcastImport < BaseModel
+  include PRXAccess
+  include Rails.application.routes.url_helpers
+
+  attr_accessor :feed, :template, :stories, :podcast, :distribution
+
+  belongs_to :account, -> { with_deleted }
+  belongs_to :series, -> { with_deleted }
+
+  before_validation :set_defaults, on: :create
+
+  def set_defaults
+    self.status ||= 'created'
+  end
+
+  def import_later
+    PodcastImportJob.perform_later self
+  end
+
+  def import
+    update_attributes(status: 'started')
+
+    # Request the RSS feed
+    get_feed
+    update_attributes(status: 'feed retrieved')
+
+    # Create the series
+    create_series_from_podcast
+    update_attributes(status: 'series created')
+
+    # Update podcast attributes
+    create_podcast
+    update_attributes(status: 'podcast created')
+
+    # Create the episodes
+    create_stories
+    update_attributes(status: 'complete')
+  rescue StandardError => err
+    update_attributes(status: 'failed')
+    raise err
+  end
+
+  def get_feed
+    response = connection.get(uri.path, uri.query_values)
+    podcast_feed = Feedjira::Feed.parse(response.body)
+    validate_feed(podcast_feed)
+    self.feed = podcast_feed
+  end
+
+  def validate_feed(podcast_feed)
+    if !podcast_feed.is_a?(Feedjira::Parser::Podcast)
+      parser = podcast_feed.class.name.demodulize.underscore.humanize rescue ''
+      raise "Failed to retrieve #{url}, not a podcast feed: #{parser}"
+    end
+  end
+
+  def create_series_from_podcast(feed = self.feed)
+    # create the series
+    self.series = create_series!(
+      account: account,
+      title: feed.title,
+      short_description: feed.itunes_subtitle,
+      description: feed.description
+    )
+    save!
+
+    # Add images to the series
+    if !feed.itunes_image.blank?
+      series.images.create!(
+        upload: feed.itunes_image,
+        purpose: Image::PROFILE
+      )
+    end
+
+    if feed.image && feed.image.url
+      series.images.create!(
+        upload: feed.image.url,
```

Comment:
Perhaps-uninformed question: how come here the upload points to the url, and for itunes image it points to the itunes image as a whole?